metadata/metadata.go

```diff
@@ -105,3 +147,41 @@ func safeCopy(destination map[string]string, key, val string) {
 		destination[key] = val
 	}
 }
+
+// copyWithNewPrefix copies key/value pairs from a source map to a destination map where the key matches the specified prefix.
+// If replacementPrefix is different from prefix, the prefix will be replaced while performing the copy.
+func copyWithNewPrefix(src, dest map[string]string, prefix, replacementPrefix string) {
+	for key, value := range src {
+		if strings.HasPrefix(key, prefix) {
+			newKey := key
+			if prefix != replacementPrefix {
+				newKey = strings.Replace(key, prefix, replacementPrefix, 1)
+			}
+			dest[newKey] = value
+		}
+	}
+}
```

Comment:
I'd probably replace this function with two new ones:
```
func copyWithPrefixReplacement(source, destination map[string]string, sourcePrefix, destinationPrefix string) {
	for key, value := range source {
		if strings.HasPrefix(key, sourcePrefix) {
			destination[strings.Replace(key, sourcePrefix, destinationPrefix, 1)] = value
		}
	}
}

func copyByPrefix(source, destination map[string]string, prefix string) {
	copyWithPrefixReplacement(source, destination, prefix, prefix)
}
```

This way we support both operations.